Documents/AnalyticsToolDesignDocument.md

```diff
@@ -0,0 +1,207 @@
+# VFS For Git: Analytics Tooling Design Document
+
+This document details the design of and plan for the **VFS4G analytics tool**. The tool aims to help users identify reasons why their enlistment may be slowing down and the `git` commands running slowly on a VFS4G mounted repository. By collecting and reporting statistics, the power to uncover and hopefully fix problems as they arrive moves to the user, empowering them to use the software to its fullest potential.
+
+## Problem
+
+Over the lifetime of an enlistment, sometimes a user's working directory fills up with too many entries. Generally this means having too many placeholders on their disk (have been read and pulled from remote) or too many modified files. This causes unacceptable losses in performance and greatly hinders users as they go about their daily development cycle. Additionally, if a local enlistment has been in use for a long time, the number of loose object may eventually reach the order of several thousand in which case that also contributes to the loss in performance.
+
+## End Goal
+
+```terminal
+$ gvfs statistics
+
+Repository statistics
+Total paths tracked by git:        3,123,456 |  100%
+Total number of placeholders:        123,456 |    4%
+Total number of modififed paths:       1,234 |   <1%
+
+Total hydration percentage:                       4%
+Most hydrated directories:
+  98% | src/main    | Primarily hydrated by "Sublime Text Editor"
+  79% | src/util    | Primarily hydrated by "Visual Studio 2019"
+  58% | assets/img  | Primarily hydrated by "Photoshop"
+

```

Comment:
Could/should we extend this to include folder placeholders as well as file placeholders?

Reply:
I think I'm going to eventually add a verbose option to the tool which will give those outputs as well. Just that in the base case I'm assuming that this is being used by someone a little higher level who may not even fully understand what a placeholder or modified path is so I was trying to present the simplest statistics first. I figured that the extra options for the tool might be a little more detailed than the scope of this document.